src/sources/http_scrape/mod.rs

```diff
@@ -0,0 +1,233 @@
+    .flatten()
+    .map(move |url| {
+        let client = HttpClient::new(inputs.tls.clone(), &inputs.proxy)
+            .expect("Building HTTP client failed");
+        let endpoint = url.to_string();
+
+        let mut context = context.clone();
+        context.build(&url);
+
+        let mut builder = Request::get(&url).header(http::header::ACCEPT, "text/plain");
+
+        // add user supplied headers
+        if let Some(headers) = &inputs.headers {
+            for header in headers {
+                builder = builder.header(header.0, header.1);
+            }
+        }
+        let mut request = builder.body(Body::empty()).expect("error creating request");
+
+        if let Some(auth) = &inputs.auth {
+            auth.apply(&mut request);
+        }
+
+        let start = Instant::now();
+        client
+            .send(request)
+            .map_err(Error::from)
+            .and_then(|response| async move {
+                let (header, body) = response.into_parts();
+                let body = hyper::body::to_bytes(body).await?;
+                emit!(EndpointBytesReceived {
+                    byte_size: body.len(),
+                    protocol: "http",
+                    endpoint: endpoint.as_str(),
+                });
+                Ok((header, body))
+            })
+            .into_stream()
+            .filter_map(move |response| {
+                ready(match response {
+                    Ok((header, body)) if header.status == hyper::StatusCode::OK => {
+                        emit!(RequestCompleted {
+                            start,
+                            end: Instant::now()
+                        });
+                        match context.on_response(&url, &header, &body) {
+                            Some(events) => {
+                                emit!(HttpScrapeEventsReceived {
+                                    byte_size: events.size_of(),
+                                    count: events.len(),
+                                    uri: url.clone()
+                                });
+
+                                // TODO the below seems wrong placement. It seems should happen
+                                // after the stream is written ? Yet I'm not seeing how to go about
+                                // that.
+                                // emit EventsSent if metrics
+                                if !events.is_empty() {
+                                    if let Event::Metric(ref _metric) =
+                                        events.first().expect("should have event")
+                                    {
+                                        emit!(HttpScrapeEventsSent {
+                                            count: events.len() as u64,
+                                            byte_size: events.size_of()
+                                        });
+                                    }
+                                }
```

Comment:
Yeah, I think we should aim to support all data types since it depends on the configured codec. Using the `native_json` codec would create logs, metrics, or traces.

Reply:
Hmm ok, I know that the source accepts all data types because the integration tests work with the different data types using `native_json` 
, but perhaps the integration tests are not validating that the correct internal_events are emitted.
That particular bit of code to only emit `EventsSent` if its a metric event, I think I grabbed from another source. But maybe its not correct to filter on metric events in this case.

Reply:
Looking at Jesse's other comment on this bit of code, sounds like I don't even need this.

Reply:
Yeah this was unnecessary, I've removed it.